spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/BlockingQueueConsumer.java

```diff
@@ -851,6 +859,28 @@ public boolean commitIfNecessary(boolean localTx) throws IOException {
 
 	}
 
+	/**
+	 * Notify MessageAckListener set on the relevant message listener.
+	 * @param messageAckListener MessageAckListener set on the message listener.
+	 * @param success Whether ack succeeded.
+	 * @param deliveryTag The deliveryTag of ack.
+	 * @param cause If an exception occurs.
+	 */
+	private void notifyMessageAckListener(@Nullable MessageAckListener messageAckListener,
+										  boolean success,
+										  long deliveryTag,
+										  @Nullable Throwable cause) {
+		if (messageAckListener == null) {
```

Comment:
We can avoid this null check by making the default `(success, tag, ex) -> { }` and `Assert.notNull(...)` in the setter (which you have in the AMLC).